channels/1ds-post-js/src/HttpManager.ts

```diff
@@ -1049,6 +1058,59 @@ export class HttpManager {
                 }
             }
 
+            function _preparePayload(callback: (processedPayload: IPayloadData) => void, payload: IPayloadData, isSync: boolean) {
+                if (_disableZip || isSync || !payload.data) {
+                    // If body is null or undefined, we don't need to compress it
+                    callback(payload);
+                    return;
+                }
+
+                const CompressionStream = (window as any).CompressionStream;
+
+                // Compression logic
+                const uint8Data = typeof payload.data === "string"
+                    ? new TextEncoder().encode(payload.data)
+                    : payload.data;
+
+                let body = new ReadableStream<Uint8Array>({
+                    start(controller) {
+                        controller.enqueue(uint8Data);
+                        controller.close();
+                    }
+                });
+    
+                const compressedStream = body.pipeThrough(new CompressionStream("gzip"));
+                const reader = (compressedStream.getReader() as ReadableStreamDefaultReader<Uint8Array>);
+                const chunks: Uint8Array[] = [];
+                let totalLength = 0;
+        
+                return reader.read().then(function processChunk({ done, value }: { done: boolean, value?: Uint8Array }): IPromise<void> {
```

Comment:
For minification you can use [`doAwaitResponse(...)`](https://nevware21.github.io/ts-async/typedoc/functions/doAwaitResponse.html) as this only then needs 1 function to be defined.

Also as the response is never used, you can drop the `return` as that won't get minified either.

Reply:
Might be easier to break this out into a local (not in-line) function or a helper function that is passed the reader that returns the function so it has access to the reader internally.